server/mailrepository/mailrepository-api/src/test/java/org/apache/james/mailrepository/MailRepositoryContract.java

```diff
@@ -419,6 +420,22 @@ default void storingMessageWithSameKeyTwiceShouldUpdateMessageAttributes() throw
         assertThat(testee.retrieve(MAIL_1)).satisfies(actual -> checkMailEquality(actual, mail));
     }
 
+
+    @Disabled("JAMES-2608")
+    @Test
+    default void storingMessageWithPerRecipientHeadersShouldAllowMultipleHeadersPerUser() throws Exception {
+
+        MailRepository testee = retrieveRepository();
+        Mail mail = createMail(MAIL_1);
+        MailAddress recipient1 = new MailAddress("rec1@domain.com");
+        mail.addSpecificHeaderForRecipient(PerRecipientHeaders.Header.builder().name("foo").value("bar").build(), recipient1);
+        mail.addSpecificHeaderForRecipient(PerRecipientHeaders.Header.builder().name("fizz").value("buzz").build(), recipient1);
+        testee.store(mail);
+
+        assertThat(testee.list()).hasSize(1);
```

Comment:
Here you could assert `contains only MAIL_1`, no?

Reply:
yes